### PR TITLE
memleak  in evhtp_free()

### PR DESCRIPTION
**Pull request by akalend**
You must free connection  listener  evconnlistener_free(evhtp->server) or it fix into evhtp_free(evhtp)

### CODE REVIEW

File: evhtp.c [hunk 1]
```diff
@@ -4205,6 +4205,11 @@ evhtp_free(evhtp_t * evhtp) {
     }
 #endif
 
+    if (evhtp->server) { 
+        evhtp_safe_free(evhtp->server_name, evconnlistener_free);
+    }
+
+
```
Comment (ellzey) [inline evhtp.c]: Double free, this is already covered.

```C
     if (evhtp->server_name) {
         evhtp_safe_free(evhtp->server_name, free);
     }
```

Comment (akalend) [inline evhtp.c]: I had error : must be evhtp->server, NOT be evhtp->server_name:

	if (evhtp->server) {
		evhtp_safe_free(evhtp->server, evconnlistener_free);
	}

Comment (ellzey) [inline evhtp.c]: pushed, good find.

Comment (akalend) [inline evhtp.c]: yes, I find the leak by valgrind. So, I finding other leaks.

**Comment by ellzey** [comment]
we get things done

**Pull request closed without merging**